src/actions/transferActions.js

```diff
@@ -572,6 +573,301 @@ async function _getTransfer (transferId: ?string, receivingId: ?string) {
+  }
+  // direct transfer destination account
+  if (transferData.destinationAccount) {
+    transferData.destinationAccount = createAccount(
+      JSON.parse(transferData.destinationAccount)
+    ).getAccountData()
+    transferData.transferMethod = 'DIRECT_TRANSFER'
+  }
+
+  const transferMethod = transferData.senderToReceiver ? 'DIRECT_TRANSFER' : 'EMAIL_TRANSFER'
+  const state = getTransferState(transferData)
+
+  return { ...transferData, transferMethod, state }
+}
+
+async function _getTxHistoryByAccount ({
+  account,
+  accountTxHistory
+}: {
+  account: AccountData,
+  accountTxHistory: ?AccountTxHistoryType
+}) {
+  const MIN_UPDATE_INTERVAL = 10 // 10 seconds
+  const MAX_TXS = 50 // maximum number of txs
+
+  // init accountTxHistory if necessary
+  accountTxHistory = accountTxHistory
+    ? accountTxHistory
+    : {
+        updated: null,
+        history: []
+      }
+
+  // do not update if MIN_UPDATE_INTERVAL has not passed
+  if (accountTxHistory.updated + MIN_UPDATE_INTERVAL >= moment().unix())
+    return { account, accountTxHistory }
+
+  let rv = {}
+
+  let txs: Array<Object> = []
+
+  // gather recent txs using third-party APIs
+  if (getCryptoPlatformType(account.cryptoType) === 'bitcoin') {
+    // get a list of txs
+    let response = await axios.get(
+      `${url.LEDGER_API_URL}/addresses/${account.address}/transactions?noToken=true`
+    )
+
+    // sort tx by received timestamp in non-increasing order
+    txs = response.data.txs.sort(
+      (txA, txB) => moment(txB['received_at']).unix() - moment(txA['received_at']).unix()
+    )
+  } else if (getCryptoPlatformType(account.cryptoType) === 'ethereum') {
+    if (isERC20(account.cryptoType)) {
+      // token tx
+      // gather erc20 txs
+      let response = await axios.get(
+        `${url.ETHERSCAN_API_URL}/api?module=account&action=tokentx&address=${
+          account.address
+        }&contractaddress=${
+          getCrypto(account.cryptoType).address
+        }&page=1&offset=${MAX_TXS}&sort=desc&apikey=${env.REACT_APP_ETHERSCAN_API_KEY}`
+      )
+      txs = response.data.result
+    } else {
+      // eth tx
+      // 1. gather normal tx
+      let response = await axios.get(
+        `${url.ETHERSCAN_API_URL}/api?module=account&action=txlist&address=${account.address}&page=1&offset=${MAX_TXS}&sort=desc&apikey=${env.REACT_APP_ETHERSCAN_API_KEY}`
+      )
+      const txsNormal = response.data.result
+
+      // 2. gather internal tx (sent from a contract)
+      response = await axios.get(
+        `${url.ETHERSCAN_API_URL}/api?module=account&action=txlistinternal&address=${account.address}&page=1&offset=${MAX_TXS}&sort=desc&apikey=${env.REACT_APP_ETHERSCAN_API_KEY}`
+      )
+      const txsInternal = response.data.result
+
+      // combine normal txs and internal txs, order in non-increasing timestamp
+      txs = [...txsNormal, ...txsInternal].sort((txA, txB) => txB - txA)
+    }
+  }
+
+  // tx always have property hash regardless of cryptoType
```

Comment:
ok...but not sure what this comment is for??

Reply:
moved